opm/simulators/flow/Main.hpp

```diff
@@ -217,7 +217,6 @@ class Main
             EclGenericVanguard::setCommunication(std::make_unique<Parallel::Communication>(new_comm));
```

Comment:
Here is what I think will happen/is happening here:
- We have two communicators (one with just global rank 0 and one with the rest.
- Both  are used to set the communicator used for the simulation
-  Hence we have one serial simulation (global rank 0) and another one (rest of the ranks) that might be parallel, both with the same arguments.

Shouldn't the two simulations be constantly fighting over the output files? I would assume that either one of them just wins or (more likely) both output will end up in the same files. Not sure how the compare script handles this.

My suggestion would be:
- Make the simulator or just the startup aware of MPI_COMM_NULL and if it is encountered no simulation and output is done
- MPI_COMM_NULL for the gobal rank 0.

BTW we need to release the created communicators at the end of the program.

Reply:
Seems like that code was there already. Still can't see how it can work.

Reply:
There are indeed to simulator "sets" (for lack of a better word), one for world rank 0 and another for the rest. But this is not a problem, as the isSimulationRank_ member is used to control behaviour. The world rank 0 "set" does nothing.

Reply:
> BTW we need to release the created communicators at the end of the program.

I agree, the code in ~Main() is just a sanity check. Will try to provide a fix.

Reply:
> I agree, the code in ~Main() is just a sanity check. Will try to provide a fix.

I retract that statement. I actually think the code in the destructor is correct (although the assert() could be discussed). The comm freed here should be identical to the one created by the split.

Reply:
Indeed.